platform/core/src/capabilities/CreationCapability.js

```diff
@@ -0,0 +1,79 @@
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define,Promise*/
+
+define(
+    ['../objects/DomainObjectImpl', 'uuid'],
+    function (DomainObjectImpl, uuid) {
+        'use strict';
+
+        /**
+         * Implements the `creation` capability. This allows new domain
+         * objects to be instantiated.
+         *
+         * @constructor
+         * @memberof platform/core
+         * @param $injector Angular's `$injector`
+         */
+        function CreationCapability($injector) {
+            this.$injector = $injector;
+        }
+
+        /**
+         * Alias of `capabilityService.getCapabilities`; handles lazy loading
+         * of `capabilityService`, since it cannot be declared as a
+         * dependency directly without creating a cycle.
+         * @private
+         */
+        CreationCapability.prototype.getCapabilities = function (model) {
+            if (!this.capabilityService) {
+                this.capabilityService =
+                    this.$injector.get('capabilityService');
+            }
+            return this.capabilityService.getCapabilities(model);
+        };
+
+        /**
+         * Instantiate a new domain object with the provided model.
+         *
+         * This domain object will have been simply instantiated; it will not
+         * have been persisted, nor will it have been added to the
+         * composition of the object which exposed this capability.
+         *
+         * @returns {DomainObject} the new domain object
+         */
+        CreationCapability.prototype.create = function (model) {
+            var id = uuid(),
+                capabilities = this.getCapabilities(model);
+            return new DomainObjectImpl(id, model, capabilities);
+        };
+
+        /**
+         * Alias of `create`.
+         * @see {platform/core.CreationCapability#create}
+         */
+        CreationCapability.prototype.invoke =
+            CreationCapability.prototype.create;
+
+        return CreationCapability;
+    }
+);
```

Comment:
I agree that this is a little funny; the rationale for doing it this way is to support multiple persistence spaces, #245. I've opened a PR for this early (#257) which has some additional changes here.

To frame this a little, some of the practical use cases for multiple persistence spaces are:

* Providing domain object hierarchies for multiple missions, where each mission's hierarchy is stored at a different URL (to allow for permissions management.)
* Separating private ("My Items") from public information ("Mission Items" or similar); again, may be stored at different URLs for simplified permissions management
* (Less immediate, may not take this approach) Providing read/write access for "domain objects" which reside in a different application (e.g. xGDS traverse plans.)

In all cases, the desired behavior is that the back-end space that you persist to depends on the location of the created object.

A case where this gets complicated is Duplicate. To avoid leaving partially-duplicated trees behind, we're persisting objects deeper in the tree first, ending with an persist of the destination of the Duplicate. As such, the location hierarchy isn't complete until the last object gets created; so, we need a short-hand to say "create this new object such that it will be persisted in the same space as this other object."

In other words, the specific details of creating a new domain object need to vary depending on where it will be created, even if it won't immediately be a child of that destination. Putting something in a capability is a handy way to allow behavior to vary per domain object. (Another constraint is introduced by the implementation approach for #245; when reading objects, you need to know which space to read from, and the easiest way to do that is to encode it in the identifier. To make this work, you need to know the space at instantiation time, because that is also where you'd generate the identifier. Sticking this in a capability makes it convenient to hide those details.)

We could also handle this in `creationService` (by changing semantics of `creationService.createObject(model, parent)` to mean the same thing as `parent.useCapability('creation', model)` would in this PR.) I tend to prefer capabilities for behavior that may vary per-object, because if these rules change (or if a different deployment wants to introduce different rules for some objects) this can be reasonably well-hidden by overriding or wrapping capabilities for objects of certain types (can also be done at the service layer via decoration, but you end up with a lot of iffing to recognize the cases you want to override, and it's just cleaner to handle that polymorphically)

Anyway, open to other options, just wanted to give some context for that decision